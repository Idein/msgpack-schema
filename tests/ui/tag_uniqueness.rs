use msgpack_schema::*;

mod serialize {
    use super::*;

    #[derive(Serialize)]
    struct S1 {
        #[tag = 0]
        x: String,
        #[tag = 0]
        y: String,
    }

    #[derive(Serialize)]
    enum E1 {
        #[tag = 0]
        V1,
        #[tag = 0]
        V2,
    }
}

mod deserialize {
    use super::*;

    #[derive(Deserialize)]
    struct S1 {
        #[tag = 0]
        x: String,
        #[tag = 0]
        y: String,
    }

    #[derive(Deserialize)]
    enum E1 {
        #[tag = 0]
        V1,
        #[tag = 0]
        V2,
    }
}

fn main() {}
