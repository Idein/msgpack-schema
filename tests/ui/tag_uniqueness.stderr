error: tag values must not be duplicate
  --> $DIR/tag_uniqueness.rs:10:9
   |
10 |         #[tag = 0]
   |         ^^^^^^^^^^

error: tag values must not be duplicate
  --> $DIR/tag_uniqueness.rs:18:9
   |
18 |         #[tag = 0]
   |         ^^^^^^^^^^

error: tag values must not be duplicate
  --> $DIR/tag_uniqueness.rs:30:9
   |
30 |         #[tag = 0]
   |         ^^^^^^^^^^

error: tag values must not be duplicate
  --> $DIR/tag_uniqueness.rs:38:9
   |
38 |         #[tag = 0]
   |         ^^^^^^^^^^
